authzed/api/v1/retryable_client.py

```diff
@@ -0,0 +1,273 @@
+
+    def retryable_bulk_import_relationships(
+        self,
+        relationships: List[Relationship],
+        conflict_strategy: ConflictStrategy,
+        timeout_seconds: Optional[int] = None
+    ):
+        """
+        Import relationships with configurable retry behavior based on conflict strategy.
+        
+        Args:
+            relationships: List of relationships to import
+            conflict_strategy: Strategy to use when conflicts are detected
+            timeout_seconds: Optional timeout in seconds for the operation
+            
+        Returns:
+            The response from the successful import operation
+            
+        Raises:
+            Exception: If the import fails and cannot be retried
+        """
+        if asyncio.iscoroutinefunction(self.BulkImportRelationships):
+            return self._retryable_bulk_import_relationships_async(
+                relationships, conflict_strategy, timeout_seconds
+            )
+        else:
+            return self._retryable_bulk_import_relationships_sync(
+                relationships, conflict_strategy, timeout_seconds
+            )
+
+    def _retryable_bulk_import_relationships_sync(
+        self,
+        relationships: List[Relationship],
+        conflict_strategy: ConflictStrategy,
+        timeout_seconds: Optional[int] = None
+    ):
+        """Synchronous implementation of retryable bulk import."""
+        timeout = timeout_seconds or DEFAULT_TIMEOUT_SECONDS
+        
+        # Create a generator function to yield requests
+        def request_iterator():
+            yield BulkImportRelationshipsRequest(relationships=relationships)
+        
+        # Try bulk import first - correctly passing the request iterator
+        try:
+            response = self.BulkImportRelationships(request_iterator(), timeout=timeout)
+            return response  # Success on first try
+        except Exception as err:
+            # Handle errors based on type and conflict strategy
+            if self._is_canceled_error(err):
+                raise err
+                
+            if self._is_already_exists_error(err) and conflict_strategy == ConflictStrategy.SKIP:
+                return None  # Skip conflicts
+                
+            if self._is_retryable_error(err) or (
+                self._is_already_exists_error(err) and 
+                conflict_strategy == ConflictStrategy.TOUCH
+            ):
+                # Retry with write_relationships_with_retry
+                return self._write_batches_with_retry_sync(relationships, timeout)
+                
+            if self._is_already_exists_error(err) and conflict_strategy == ConflictStrategy.FAIL:
+                raise ValueError("Duplicate relationships found")
+                
+            # Default case - propagate the error
+            raise ValueError(f"Error finalizing write of {len(relationships)} relationships: {err}")
```

Comment:
```suggestion
            raise ValueError(f"Error finalizing write of {len(relationships)} relationships") from err
```
This will preserve the error context/stack trace.